src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -421,6 +418,7 @@ def csv(self, path: str) -> DataFrame:
                 schema_to_cast = [("$1", "C1")]
                 transformations = []
         else:
+            self._cur_options["INFER_SCHEMA"] = False
```

Comment:
this condition will already be true if we reach the else branch, right?

Reply:
No, if we reach here, user must have provide a schema, in that way, we should use what user provide and turn off infer_schema